src/benchmarklib/benchmark_sql_executor.hpp

```diff
@@ -24,23 +23,28 @@ class BenchmarkSQLExecutor {
   std::pair<SQLPipelineStatus, std::shared_ptr<const Table>> execute(
       const std::string& sql, const std::shared_ptr<const Table>& expected_result_table = nullptr);
 
+  // If auto-commit is disabled, explicitly commit / roll back the transaction
+  void commit();
+  void rollback();
+
   // Contains one entry per executed SQLPipeline
   std::vector<SQLPipelineMetrics> metrics;
 
   bool any_verification_failed = false;
 
+  // Can optionally be set by the caller. Otherwise, pipelines are auto-committed
+  std::shared_ptr<TransactionContext> transaction_context = nullptr;
+
  private:
   void _compare_tables(const std::shared_ptr<const Table>& expected_result_table,
                        const std::shared_ptr<const Table>& actual_result_table,
                        const std::optional<const std::string>& description = std::nullopt);
   void _verify_with_sqlite(SQLPipeline& pipeline);
   void _visualize(SQLPipeline& pipeline) const;
 
-  const bool _enable_jit;
-  const std::shared_ptr<SQLiteWrapper> _sqlite_wrapper;
-  const std::optional<std::string> _visualize_prefix;
-
-  const std::shared_ptr<TransactionContext> _transaction_context;
+  bool _enable_jit;
```

Comment:
Why was the const removed for these member variables?

Reply:
Leftover.